tests/tagged_vector.cpp

```diff
@@ -73,3 +73,39 @@ TEST(TaggedVector, Operators)
     ASSERT_EQ(4 + c, (ddc::detail::TaggedVector<int, double>(5)));
     ASSERT_EQ(4 * a, (ddc::detail::TaggedVector<int, double, float>(4, 8)));
 }
+
+TEST(TaggedVector, Assignment)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, double, float> b;
+    b = a;
+    EXPECT_EQ(a.get<double>(), b.get<double>());
+    EXPECT_EQ(a.get<float>(), b.get<float>());
+}
+
+TEST(TaggedVector, ReorderingAssignment)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, float, double> b;
+    b = a;
+    EXPECT_EQ(a.get<double>(), b.get<double>());
+    EXPECT_EQ(a.get<float>(), b.get<float>());
+}
+
+TEST(TaggedVector, MoveAssignment)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, double, float> b;
+    b = std::move(a);
+    EXPECT_EQ(a.get<double>(), b.get<double>());
+    EXPECT_EQ(a.get<float>(), b.get<float>());
```

Comment:
I agree that in the case of type `int` it shall work but we should not be using `a` after a move. I suggest:
```suggestion
    EXPECT_EQ(1, b.get<double>());
    EXPECT_EQ(2, b.get<float>());
```

Reply:
Oops, I didn't really think. Just copy-pasted

Reply:
Fixed [628025c](https://github.com/CExA-project/ddc/pull/210/commits/628025c3db4b224d4ce4584d299f19c4ef90c91c)